catalog/models.py

```diff
@@ -204,35 +210,80 @@ def __str__(self):
         )
 
 
+def folder_path(instance, filename):
+    """Генерирует имя файла. Возвращает путь к нему."""
+    id = instance.real_estate.pk
+    return (f'real_estate/{id}_'
+            f'{datetime.timestamp(datetime.now())*1000:.0f}.jpg')
+
+
 class Image(models.Model):
     """1toМ Модель для фотографий объекта."""
-
     real_estate = models.ForeignKey(
         RealEstate,
         on_delete=models.CASCADE,
         related_name='images',
         db_index=True,
         verbose_name='Объект',
     )
-    image = models.ImageField(upload_to='real_estate', verbose_name='Фото')
+    image = ResizedImageField(
+        verbose_name='Фото',
+        upload_to=folder_path,
+    )
 
     class Meta:
         verbose_name = 'Фотография'
         verbose_name_plural = 'Фотографии'
         ordering = ('-id',)
 
+    def image_generator(self, infile, outfile, image_size):
+        """
+        Генерирует изображения в требуемых размерах.
+        Обрезает и центрует вертикальные изображения.
+        """
+        with PillowImage.open(infile) as im:
+            width, height = im.size
+            if width <= height:
+                im_new = im.crop(
+                    (0, 0 + height // 5, width, height - height // 5)
+                )
+                im_resize = im_new.resize(image_size)
+                im_resize.save(outfile, 'JPEG')
+            else:
+                im_resize = im.resize(image_size)
+                im_resize.save(outfile, 'JPEG')
+    
+    def filename_generator(self, filepath, size):
+        weith, higth = size
+        name, file_format = filepath.split('.')
+        return f'{name}_{weith}x{higth}.{file_format}'
+
     def save(self, *args, **kwargs):
+        """Сохраняет дополнительно изображения в требуемых размерах."""
+        infile = self.image
         if (
             Image.objects.filter(real_estate=self.real_estate).count()
             >= settings.IMAGE_LIMIT
         ):
             return  # Не сохраняем, если уже 6 фото
         super(Image, self).save(*args, **kwargs)
+        for size in settings.ALL_SIZES:
```

Comment:
тут будет другая константа: PREVIEW_SIZES
```suggestion
        for size in settings.PREVIEW_SIZES:
```